python/covid_graphs/covid_graphs/predictions.py

```diff
@@ -241,36 +243,50 @@ def load_prediction_db(prediction_dir: Path) -> PredictionDb:
         ]
         # TODO(miskosz): The decision on which predictions to display should not be a reponsibility
         # of `predictions` module. All data should be served.
-        start_idx = _calculate_earliest_displayable_idx(fitted_formulas, MAX_PEAK_DISTANCE)
+        start_idx, end_idx = _calculate_displayable_interval(
+            fitted_formulas, MAX_PEAK_DISTANCE, MAX_PEAK_VARIABILITY
+        )
         fitted_predictions = _create_predictions_from_formulas(
-            fitted_formulas[start_idx:], country_atg_parameters.short_country_name
+            fitted_formulas[start_idx:end_idx], country_atg_parameters.short_country_name
         )
         country_predictions.extend(fitted_predictions)
 
     return PredictionDb(country_predictions=country_predictions)
 
 
-def _calculate_earliest_displayable_idx(
-    fitted_formulas: Iterable[FittedFormula], max_peak_distance: datetime.timedelta
-) -> int:
+def _calculate_displayable_interval(
+    fitted_formulas: Iterable[FittedFormula],
+    max_peak_distance: datetime.timedelta,
+    max_peak_variability: datetime.timedelta,
+) -> Tuple[int, int]:
     """
-    Calculates the earliest index in fitted_formulas, after which the peaks of formulas are within
-    'max_peak_distance'.
+    Returns a half-open interval [start, end), for which the following holds:
+    * fitted_formulas[end - 1] is the latest prediction whose peak is not too far in the future,
+      i.e. it's earlier than 'max_peak_distance' from now.
+    * fitted_formula[start] is the earliest prediction, after which the peaks are within
+      'max_peak_variability'.
     """
     peak_days = [
         formula.get_peak(country_report=None) for formula in reversed(list(fitted_formulas))
     ]
+    skip = 0
+    latest_peak = datetime.datetime.now() + max_peak_distance
+    while skip < len(peak_days) and peak_days[skip] > latest_peak:
+        skip += 1
+    clipped_peaks = peak_days[skip:]
+    if len(clipped_peaks) == 0:
+        return skip, skip
```

Comment:
Nit: Consider `if skip == len(peak_days):` on the line before clipping.

Reply:
I've rewrote the code, so I'm not sure if this comment still applies.